src/liballoc/string.rs

```diff
@@ -2142,6 +2150,7 @@ pub trait ToString {
 /// since `fmt::Write for String` never returns an error itself.
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T: fmt::Display + ?Sized> ToString for T {
+    #[must_use]
```

Comment:
The `#[must_use]` attribute needs to be in the trait definition. It doesn't work when added to impls: #48486.